src/components/contents/DetailViewItems/DetailView.jsx

```diff
@@ -0,0 +1,179 @@
+import { useState, useContext } from "react";
+import { useQuery, useMutation, useQueryClient } from "@tanstack/react-query";
+import PropTypes from "prop-types";
+
+import UserContext from "../../../context/UserContext";
+import CurrentDBIdContext from "../../../context/CurrentDBIdContext";
+import FieldList from "./FieldList";
+
+import CONSTANT from "../../../constants/constant";
+
+import fetchData from "../../../utils/axios";
+
+const { X_DRAG_ADJUSTMENT, Y_DRAG_ADJUSTMENT } = CONSTANT;
+
+function DetailView({
+  isEditMode,
+  setIsEditMode,
+  currentDocIndex,
+  documentsIds,
+  isOnSave,
+  setIsOnSave,
+}) {
+  const [docData, setDocData] = useState([]);
+  const [isDragging, setIsDragging] = useState(false);
+  const [draggedElementIndex, setDraggedElementIndex] = useState(null);
+
+  const queryClient = useQueryClient();
+
+  const { userId } = useContext(UserContext);
+  const currentDBId = useContext(CurrentDBIdContext);
+
+  async function getDocument() {
+    const response = await fetchData(
+      "GET",
+      `users/${userId}/databases/${currentDBId}/documents/${documentsIds[currentDocIndex]}`,
+    );
+
+    return response;
+  }
+
+  const { isLoading } = useQuery(["document", currentDocIndex], getDocument, {
+    enabled: !!userId && !!documentsIds,
+    onSuccess: result => {
+      const receivedFields = result.data.document.fields.map(element => {
+        return {
+          fieldName: element.fieldName,
+          fieldType: element.fieldType,
+          fieldValue: element.fieldValue,
+          xCoordinate: element.xCoordinate,
+          yCoordinate: element.yCoordinate,
+          fieldId: element._id,
+        };
+      });
+      setDocData(receivedFields);
+    },
+    onFailure: () => {
+      console.log("sending user to errorpage");
+    },
+    refetchOnWindowFocus: false,
+  });
+
+  async function handleClickSave() {
+    await fetchData(
+      "PUT",
+      `/users/${userId}/databases/${currentDBId}/documents/${documentsIds[currentDocIndex]}`,
+      { fields: docData },
+    );
+  }
+
+  const { mutate: fetchDocumentUpdate } = useMutation(handleClickSave, {
+    onSuccess: () => {
+      queryClient.refetchQueries(["dbDocumentList"]);
+    },
+    onFailure: () => {
+      console.log("sending user to errorpage");
+    },
+    refetchOnWindowFocus: false,
+  });
+
+  if (isLoading) {
+    return <h1>Loading</h1>;
+  }
+
+  if (!isEditMode && isOnSave) {
+    fetchDocumentUpdate();
+    setIsOnSave(false);
+  }
+
+  if (isLoading) {
+    return <h1>Loading</h1>;
+  }
```

Comment:
```suggestion
  if (isLoading) {
    return <h1>Loading</h1>;
  }

  if (!isEditMode && isOnSave) {
    fetchDocumentUpdate();
    setIsOnSave(false);
  }
```

isLoading 시에 Loading 을 반납하는 함수가 중복되어 있습니다! 
수정 부탁드리겠습니다!

Reply:
🤕!! 수정하였습니다!!